### ISSUE TITLE
F#'s "Find references" wrongly returns "0 matches" too often

### ISSUE THREAD

**Issue by knocte** [mono/monodevelop#4482]
Steps to reproduce:
1. Clone https://github.com/knocte/gwallet/
2. Open it in VS4Mac (stable channel).
3. Go to file ElectrumServer.fs.
4. Right click defaultBtcList -> Find references.

Current results: 0 matches
Expected results: 1 or 2 matches.

Version info:
```
=== Visual Studio Community 2017 for Mac ===

Version 7.4.1 (build 48)
Installation UUID: fd06ff65-69de-47ae-8b99-ddf257032b7c
Runtime:
	Mono 5.8.1.0 (2017-10/6bf3922f3fd) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)

	Package version: 508010000

=== NuGet ===

Version: 4.3.1.4445

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.0.5
	2.0.0
SDK: /usr/local/share/dotnet/sdk/2.1.4/Sdks
SDK Versions:
	2.1.4
	2.0.0
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.8.1/lib/mono/msbuild/15.0/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.1
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Apple Developer Tools ===

Xcode 9.2 (13772)
Build 9C40b

=== Xamarin.Mac ===

Version: 4.2.1.28 (Visual Studio Community)

=== Xamarin.iOS ===

Version: 11.8.1.28 (Visual Studio Community)
Hash: 3979d081
Branch: d15-6
Build date: 2018-03-09 16:22:34-0500

=== Xamarin.Android ===

Version: 8.2.0.16 (Visual Studio Community)
Android SDK: /Users/andres/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		4.4 (API level 19)
		7.1 (API level 25)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 26.0.2
SDK Build Tools Version: 26.0.3

Java SDK: /usr
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Xamarin Inspector ===

Version: 1.4.0
Hash: b3f92f9
Branch: master
Build date: Fri, 19 Jan 2018 22:00:34 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 704010048
Git revision: 6a86203dd590c6cfdf814c8832d24d2ad1e2606f
Build date: 2018-03-20 11:16:55-04
Xamarin addins: 8d83cea9e4ae03cb0c14ba3fbaafdbb0cbd170e7
Build lane: monodevelop-lion-d15-6

=== Operating System ===

Mac OS X 10.13.1
Darwin 17.2.0 Darwin Kernel Version 17.2.0
    Fri Sep 29 18:27:05 PDT 2017
    root:xnu-4570.20.62~3/RELEASE_X86_64 x86_64

=== Enabled user installed addins ===

Internet of Things (IoT) development (Preview) 7.1


```

> VS bug [#597602](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/597602)

**Comment by nosami** [comment]
![gwalletfindrefs](https://user-images.githubusercontent.com/667194/40429927-be29b4de-5e9b-11e8-9aec-2f67951a8506.gif)

This fix seems to be working pretty well. Previously it was checking all dependent projects when there was no need for a local symbol.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-14] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #4879 merged by: Jason Imison  
 Author: mono  
 Commit: 4eca9664495dfc65d6b5b055135eac5d344d5349 (xamarin/monodevelop)
